pkg/cloudprovider/suite_test.go

```diff
@@ -126,269 +111,112 @@ var _ = BeforeEach(func() {
-		cloudProviderMachine, err := cloudProvider.Create(ctx, nodeClaim)
-		Expect(corecloudprovider.IsInsufficientCapacityError(err)).To(BeTrue())
-		Expect(cloudProviderMachine).To(BeNil())
-	})
+// Helper functions for NodeClaim validation
+func validateNodeClaimCommon(nodeClaim *karpv1.NodeClaim, nodePool *karpv1.NodePool) {
+	// Basic validation
+	Expect(nodeClaim).ToNot(BeNil())
+	Expect(nodeClaim.Status.Capacity).ToNot(BeEmpty())
+
+	// Status fields
+	Expect(nodeClaim.Status.ProviderID).ToNot(BeEmpty())
+	Expect(nodeClaim.Status.ImageID).ToNot(BeEmpty())
+
+	// Common labels validation
+	Expect(nodeClaim.Labels).To(HaveKey(karpv1.CapacityTypeLabelKey))
+	Expect(nodeClaim.Labels).To(HaveKey(karpv1.NodePoolLabelKey))
+	if nodePool != nil {
+		Expect(nodeClaim.Labels[karpv1.NodePoolLabelKey]).To(Equal(nodePool.Name))
+	}
+	Expect(nodeClaim.Labels).To(HaveKey(v1.LabelInstanceTypeStable))
+	Expect(nodeClaim.Labels).To(HaveKey(v1.LabelArchStable))
+	Expect(nodeClaim.Labels).To(HaveKey(v1beta1.LabelSKUName))
+	Expect(nodeClaim.Labels).To(HaveKey(v1beta1.LabelSKUFamily))
+	Expect(nodeClaim.Labels).To(HaveKey(v1beta1.LabelSKUCPU))
+	Expect(nodeClaim.Labels).To(HaveKey(v1beta1.LabelSKUMemory))
+
+	// Zone validation (conditional)
+	if nodeClaim.Labels[v1.LabelTopologyZone] != "" {
+		Expect(nodeClaim.Labels[v1.LabelTopologyZone]).To(MatchRegexp(`^[a-z0-9-]+-[0-9]+$`))
+	}
+
+	// Capacity and Allocatable resources
+	Expect(nodeClaim.Status.Capacity).To(HaveKey(v1.ResourceCPU))
+	Expect(nodeClaim.Status.Capacity).To(HaveKey(v1.ResourceMemory))
+	Expect(nodeClaim.Status.Allocatable).To(HaveKey(v1.ResourceCPU))
+	Expect(nodeClaim.Status.Allocatable).To(HaveKey(v1.ResourceMemory))
+
+	// Lifecycle validation
+	Expect(nodeClaim.DeletionTimestamp).To(BeNil())
+}
 
-	// TODO (chmcbrid): split Drift tests into their own test file drift_test.go
-	Context("Drift", func() {
-		var nodeClaim *karpv1.NodeClaim
-		var pod *v1.Pod
-		var node *v1.Node
+func validateVMNodeClaim(nodeClaim *karpv1.NodeClaim, nodePool *karpv1.NodePool) {
+	// Common validations
+	validateNodeClaimCommon(nodeClaim, nodePool)
+}
 
+var _ = Describe("CloudProvider", func() {
+	// Attention: tests under "ProvisionMode = AKSScriptless" are not applicable to ProvisionMode = AKSMachineAPI option.
+	// Due to different assumptions, not all tests can be shared. Add tests for AKS machine instances in a different Context/file.
+	// If ProvisionMode = AKSScriptless is no longer supported, their code/tests will be replaced with ProvisionMode = AKSMachineAPI.
+	Context("ProvisionMode = AKSScriptless", func() {
 		BeforeEach(func() {
-			instanceType := "Standard_D2_v2"
-			ExpectApplied(ctx, env.Client, nodePool, nodeClass)
-			pod = coretest.UnschedulablePod(coretest.PodOptions{
-				NodeSelector: map[string]string{v1.LabelInstanceTypeStable: instanceType},
+			testOptions = test.Options(test.OptionsFields{
+				ProvisionMode: lo.ToPtr(consts.ProvisionModeAKSScriptless),
 			})
```

Comment:
Ok, so this is the proposed mechanism for executing provision mode-specific tests in their respective provision mode? I don't see this applied in other tests ... Also, this set of tests was presumably working for bootstrapping client as well?